Modules/Filtering/ImageFeature/test/itkZeroCrossingBasedEdgeDetectionImageFilterTest.cxx

```diff
@@ -20,35 +20,71 @@
 #include "itkZeroCrossingBasedEdgeDetectionImageFilter.h"
 #include "itkNullImageToImageFilterDriver.hxx"
 #include "itkSimpleFilterWatcher.h"
+#include "itkTestingMacros.h"
 
 int
-itkZeroCrossingBasedEdgeDetectionImageFilterTest(int, char *[])
+itkZeroCrossingBasedEdgeDetectionImageFilterTest(int argc, char * argv[])
 {
-  try
+  if (argc != 3)
   {
-    using ImageType = itk::Image<float, 2>;
-
-    // Set up filter
-    itk::ZeroCrossingBasedEdgeDetectionImageFilter<ImageType, ImageType>::Pointer filter =
-      itk::ZeroCrossingBasedEdgeDetectionImageFilter<ImageType, ImageType>::New();
-
-    itk::SimpleFilterWatcher watcher(filter);
-    filter->SetVariance(1.0f);
-    filter->SetMaximumError(.01f);
-
-    // Run Test
-    itk::Size<2> sz;
-    sz[0] = 100;
-    sz[1] = 100;
-    itk::NullImageToImageFilterDriver<ImageType, ImageType> test1;
-    test1.SetImageSize(sz);
-    test1.SetFilter(filter);
-    test1.Execute();
+    std::cerr << "Missing Parameters." << std::endl;
+    std::cerr << "Usage: " << itkNameOfTestExecutableMacro(argv) << " varianceValue maximumErrorValue" << std::endl;
+    return EXIT_FAILURE;
+  }
+
+  using ImageType = itk::Image<float, 2>;
+
+  // Set up filter
+  using FilterType = itk::ZeroCrossingBasedEdgeDetectionImageFilter<ImageType, ImageType>;
+
+  FilterType::Pointer filter = FilterType::New();
+
+  ITK_EXERCISE_BASIC_OBJECT_METHODS(filter, ZeroCrossingBasedEdgeDetectionImageFilter, ImageToImageFilter);
+
+
+  itk::SimpleFilterWatcher watcher(filter);
+
+  float varianceValue = std::stod(argv[1]);
+  filter->SetVariance(varianceValue);
+  for (auto i : filter->GetVariance())
+  {
+    std::cerr << "Test failed!" << std::endl;
+    std::cerr << "Error in itk::ZeroCrossingBasedEdgeDetectionImageFilter::GetVariance" << std::endl;
+    std::cerr << "Expected: " << varianceValue << ", but got: " << i << std::endl;
+    return EXIT_FAILURE;
   }
-  catch (const itk::ExceptionObject & err)
+
+  FilterType::ArrayType variance;
+  variance.Fill(varianceValue);
+  filter->SetVariance(variance);
+  ITK_TEST_SET_GET_VALUE(variance, filter->GetVariance());
+
+  float maximumErrorValue = std::stod(argv[2]);
+  filter->SetMaximumError(maximumErrorValue);
+  for (auto i : filter->GetMaximumError())
   {
-    (&err)->Print(std::cerr);
+    std::cerr << "Test failed!" << std::endl;
```

Comment:
Same, but it doesn't get here.

Reply:
Yep, my bad. Thanks @dzenanz. Addressed in 7123bdc.